itests/deals_anycid_test.go

```diff
@@ -0,0 +1,143 @@
+package itests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/lotus/api"
+
+	ipld "github.com/ipfs/go-ipld-format"
+
+	"github.com/filecoin-project/go-fil-markets/shared"
+
+	"github.com/ipld/go-car"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/node/modules/dtypes"
+
+	"github.com/filecoin-project/go-fil-markets/storagemarket"
+	"github.com/filecoin-project/lotus/node"
+	"github.com/filecoin-project/lotus/node/modules"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/actors/policy"
+	"github.com/filecoin-project/lotus/itests/kit"
+	logging "github.com/ipfs/go-log/v2"
+)
+
+func TestDealRetrieveByAnyCid(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+
+	ctx := context.Background()
+	oldDelay := policy.GetPreCommitChallengeDelay()
+	policy.SetPreCommitChallengeDelay(5)
+	t.Cleanup(func() {
+		policy.SetPreCommitChallengeDelay(oldDelay)
+	})
+
+	// Allow 8MB sectors
+	kit.EnableLargeSectors(t)
+	policy.SetSupportedProofTypes(abi.RegisteredSealProof_StackedDrg8MiBV1)
+	kit.QuietMiningLogs()
+	//logging.SetLogLevel("dagstore.invidx", "debug")
+	logging.SetLogLevel("markets-rtvl", "debug")
+	logging.SetLogLevel("markets-rtvl-reval", "debug")
+
+	// For these tests where the block time is artificially short, just use
+	// a deal start epoch that is guaranteed to be far enough in the future
+	// so that the deal starts sealing in time
+	startEpoch := abi.ChainEpoch(2 << 12)
+
+	// Override the dependency injection for the blockstore accessor, so that
+	// we can get a reference to the blockstore containing our deal later in
+	// the test
+	var bsa storagemarket.BlockstoreAccessor
+	bsaFn := func(importmgr dtypes.ClientImportMgr) storagemarket.BlockstoreAccessor {
+		bsa = modules.StorageBlockstoreAccessor(importmgr)
+		return bsa
+	}
+	bsaOpt := kit.ConstructorOpts(node.Override(new(storagemarket.BlockstoreAccessor), bsaFn))
+
+	// Allow 512MB sectors
+	eightMBSectorsOpt := kit.SectorSize(8 << 20)
+
+	// Create a client, and a miner with its own full node
+	_, client, miner, ens := kit.EnsembleTwoOne(t, kit.MockProofs(), bsaOpt, eightMBSectorsOpt)
```

Comment:
Note here for @raulk and @aarshkshah1992  - we normally use the `EnsembleMinimal` which re-uses full nodes between `miner` and `client` - I believe this is one (if not the only) example, which tests deals properly, as we are NOT reusing the full node between the miner and the client.